### PR TITLE
feat(lock): count lock for tab name

### PR DESCRIPTION
**Pull request by stonebuzz**
Count Lock for current item 
and at it to tab name

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by cedric-anne — APPROVED** [review]

**Comment by trasher** [comment]
This counts only `LockedFields` from inventory, while the tab display way more... That does not seems totally correct to me; I'll wait for @orthagh review.

**Comment by orthagh** [comment]
The counter should reflect what is displayed into the tab, global or not.
UI should be predictable and not confuse user.

**Comment by stonebuzz** [comment]
So weed to count (only for computer)

- glpi_computers_items (```where is_deleted = 1```)
- item_disks (```where is_deleted = 1```)
- item_softwareversion (```where is_deleted = 1```)
- item_softwarelicense (```where is_deleted = 1```)
- networkport (```where is_deleted = 1```)
- networkname (```where is_deleted = 1```)
- ipaddress (```where is_deleted = 1```)
- item_devices (```where is_deleted = 1```)

**Review by g-bougard — APPROVED** [review]

**Comment by trasher** [comment]
I do not understand why only deleted?

**Comment by stonebuzz** [comment]
> I do not understand why only deleted?

From tab content, GLPI display 

- All ```LockField``` (global or not) for current ```item```
- All related link (```SofwareVersion```, ```SofwareLicense``` etc ..) which are deleted (but not yet purged)

If a link  is deleted by user is for is to indicate that he no longer wishes to see it brought up from the dynamic inventory.

Best regards

**Comment by trasher** [comment]
OK, this is the same queries as `showForItem`. I guess this may be factorized

File: src/Lock.php [hunk 1]
```diff
@@ -535,9 +705,9 @@ public static function showForItem(CommonDBTM $item)
      **/
     public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0)
     {
-
         if ($item->isDynamic() && $item->can($item->fields['id'], UPDATE)) {
-            return Lock::getTypeName(Session::getPluralNumber());
+            $nb = Lock::countLockFromItem($item);
```
Comment (cedric-anne) [inline src/Lock.php:709]: All these queries should not be done if user does not want to show counters on tabs.
```suggestion
            $nb = ($_SESSION['glpishow_count_on_tabs']) ? Lock::countLockFromItem($item) : 0;
```

File: src/Lock.php [hunk 2]
```diff
@@ -526,6 +526,176 @@ public static function showForItem(CommonDBTM $item)
...
+                ]
+            ],
+            'WHERE'     => [
+                'isv.is_deleted'  => 1,
+                'isv.is_dynamic'  => 1,
+                'isv.items_id'     => $item->fields['id'],
+                'isv.itemtype'     => $item->getType(),
+            ]
+        ]);
+        $nb += $iterator->count();
```
Comment (cedric-anne) [inline src/Lock.php:664]: On this one, I do not get why there are joined tables, as there is no WHERE condition on them.

File: src/Lock.php [hunk 3]
```diff
@@ -526,6 +526,176 @@ public static function showForItem(CommonDBTM $item)
...
+                    ]
+                ]
+            ],
+            'WHERE'     => [
+                'isl.is_deleted'  => 1,
+                'isl.is_dynamic'  => 1,
+                'isl.items_id'    => $item->fields['id'],
+                'isl.itemtype'    => $item->getType(),
+            ]
+        ]);
```
Comment (cedric-anne) [inline src/Lock.php:694]: Same remark here, and result is not even used.

**Review by cedric-anne — COMMENTED** [review]
Not sure it is a good idea to do this. How many queries it adds, for just a counter ?

File: src/Lock.php (hunk 2 above)
Comment (stonebuzz) [reply src/Lock.php]: done !

File: src/Lock.php (hunk 3 above)
Comment (stonebuzz) [reply src/Lock.php]: done !

**Comment by cedric-anne** [comment]
How many queries it adds, on a computer that has many devices ?

**Comment by stonebuzz** [comment]
> Not sure it is a good idea to do this. How many queries it adds, for just a counter ?

At least 8 queries 

- glpi_lockedfields
- glpi_computers_items (where is_deleted = 1) (only for computer)
- computer VM
- item_disks (where is_deleted = 1)
- item_softwareversion (where is_deleted = 1)
- item_softwarelicense (where is_deleted = 1)
- networkport (where is_deleted = 1)
- item_devices (where is_deleted = 1)

and more for 
- ```NetworkName```
- ```IPAddress```

**Comment by cedric-anne** [comment]
I count at least 28 queries, and many more if asset has some network ports.

**Comment by stonebuzz** [comment]
As seen together, the large number of requests does not satisfy the performance to just add a ```count```

**Pull request closed without merging**